src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -138,12 +181,12 @@ private static bool IsDeclarationStartContext(List<SyntaxBase> matchingNodes, in
                         // 2. the token is a newline
                         return token.Type == TokenType.EndOfFile || 
                                token.Type == TokenType.NewLine;
-                    
-                    case SkippedTriviaSyntax _:
-                        // we are in a partial declaration
+
+                    case SkippedTriviaSyntax _ when matchingNodes.Count >= 3:
```

Comment:
This condition was inaccurate and caused declaration completions to show up in wrong places.